parachain/pallets/basic-channel/src/inbound/envelope.rs

```diff
@@ -22,6 +23,8 @@ pub struct Envelope {
 	pub channel: H160,
 	/// The application on Ethereum where the message originated from.
 	pub source: H160,
+	/// The user on Ethereum that authorized the source to send the message.
+	pub user: H160,
```

Comment:
This maps to `origin` on the [BasicOutboundChannel](https://github.com/Snowfork/snowbridge/pull/687/files#diff-0c18c33e86533e5721bc116be59e03b14f120dc10c751f4136a99547fdc56cb4R20) end. Why did we change the change name on the parachain side to `user`?

Reply:
This came from a [previous comment](https://github.com/Snowfork/snowbridge/pull/687#discussion_r972933413). My understanding is that `origin` has a slightly different meaning [in Substrate](https://docs.substrate.io/reference/glossary/#origin): "... the source of a dispatched function call...", which corresponds to the `source` field here in the envelope or the `operator` in contract authorization. The `user` field here is the address that authorized the message but not necessarily the one that sent it, which is the `origin` in contract authorization.

As I understand it:
- Substrate origin == Parachain envelope source == Contract operator
- Parachain envelope user == Contract origin

Our dispatch pallet already [converts the source to an origin](https://github.com/Snowfork/snowbridge/blob/main/parachain/pallets/dispatch/src/lib.rs#L121), so avoiding using `origin` in the envelope (for a Contract origin) avoids confusing it with a Substrate origin.

cc @vgeddes please keep me honest here.